### PR TITLE
feat(as-5748): add final comments page

### PR DESCRIPTION
**Pull request by james-hill-kc**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-5748

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/validators/final-comment/final-comment.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const { body } = require('express-validator');
+const {
+	rule: doesNotIncludeSensitiveInformationRule
+} = require('../common/does-not-include-sensitive-information');
+const { numberToLocaleString } = require('../../lib/number-functions');
+
+const maxCharacters = 32500;
```
Comment (edpackard-work) [inline packages/forms-web-app/src/validators/final-comment/final-comment.js:7]: could/should this go in config file?

**Review by edpackard-work — DISMISSED** [review]

**Comment by guiribeirokc** [comment]
Where are the tests at least for the router and the number function created?

**Review by guiribeirokc — APPROVED** [review]

File: packages/forms-web-app/src/validators/final-comment/final-comment.js (hunk 1 above)
Comment (james-hill-kc) [reply packages/forms-web-app/src/validators/final-comment/final-comment.js:7]: Just checking, would this be best placed in `src/configuration/config.js`?  And if so should we be pulling the value from an environment variable?

**Comment by james-hill-kc** [comment]
@guiribeirokc good shout on the unit test for the number-functions method, I've added this now. I also added a test for the route, although as mentioned I think there's room for debate around the value of unit testing routes/controllers etc. But this is a discussion for another time.

File: packages/forms-web-app/src/validators/final-comment/final-comment.js (hunk 1 above)
Comment (edpackard-work) [reply packages/forms-web-app/src/validators/final-comment/final-comment.js:7]: you could put it there, and do a `key: environment variable || default value` sort of thing, with the default being 32500?

File: packages/forms-web-app/src/views/final-comment/final-comment.njk [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          titleText: "There is a problem",
+          errorList: errorSummary,
+          attributes: {"data-cy": "error-wrapper"}
+        }) }}
+      {% endif %}
+      <form action="" method="post" novalidate>
+				<h1 class="govuk-heading-l">
+					What is your final comment?
+				</h1>
+        <p>Your final comment should only respond to the statements from the ((local planning authority / appellant)) or interested parties, and must not include any new evidence.</p>
```
Comment (edpackard-work) [inline packages/forms-web-app/src/views/final-comment/final-comment.njk:26]: confirm with design what is required here?

File: packages/forms-web-app/src/validators/final-comment/final-comment.js (hunk 1 above)
Comment (james-hill-kc) [reply packages/forms-web-app/src/validators/final-comment/final-comment.js]: Done!

File: packages/forms-web-app/src/views/final-comment/final-comment.njk (hunk 2 above)
Comment (james-hill-kc) [reply packages/forms-web-app/src/views/final-comment/final-comment.njk]: I've confirmed with design, the final comments page should use the "local planning authority" text. The "appellant" text is an alternative version intended for future use on the "LPA" journey. I've updated the template accordingly.

**Review by edpackard-work — DISMISSED** [review]

File: packages/forms-web-app/src/views/final-comment/final-comment.njk [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{# todo: change layout based on appeal type when component reused #}
+{% extends "layouts/full-appeal-banner/main.njk" %}
+
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
+{% from "govuk/components/textarea/macro.njk" import govukTextarea %}
+{% from "govuk/components/checkboxes/macro.njk" import govukCheckboxes %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+
+{% set title = "Your final comments - Submit final comments - Appeal a planning decision - GOV.UK" %}
```
Comment (edpackard-work) [inline packages/forms-web-app/src/views/final-comment/final-comment.njk:9]: should this be Submit final comment

File: packages/forms-web-app/src/controllers/final-comment/final-comment.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+			errors,
+			errorSummary
+		});
+	}
+
+	try {
+		req.session.finalComment.finalComment = finalComment;
+		req.session.finalComment.doesNotIncludeSensitiveInformation = !hasSensitiveInformation;
+
+		return res.redirect(`/full-appeal/submit-final-comments/documents-check`);
```
[the remark this answers is not in the document]
Comment (james-hill-kc) [reply packages/forms-web-app/src/controllers/final-comment/final-comment.js]: Thanks, I've updated this now

File: packages/forms-web-app/src/views/final-comment/final-comment.njk (hunk 3 above)
Comment (james-hill-kc) [reply packages/forms-web-app/src/views/final-comment/final-comment.njk]: Updated (also updated the validation error message from "final comments" to "final comment"

**Review by edpackard-work — APPROVED** [review]

**Pull request merged**